metacatalog/ext/standard_export/extension.py

```diff
@@ -0,0 +1,109 @@
+from metacatalog import api, cmd
+from metacatalog.models import Entry
+from metacatalog.util.results import ImmutableResultSet
+
+
+class StandardsExportExtension(MetacatalogExtensionInterface):
+    r"""
+    Extension to export Entries in standard format.
+    Currently, ISO 19115 export is implemented.
+
+    Adds the method export_iso19115 to :class:`Entry <metacatalog.models.Entry>`
+    which creates the ISO 19115 XML for the :class:`ImmutableResultSet <metacatalog.utils.results.ImmutableResultSet>`
+    of the Entry.
+    The method create_iso19115 is added to the API
+    (metacatalog.api.catalog). This method can be
+    used to export all Entries / ImmutableResultSets
+    in the database session and write the XML files
+    to the folder location specified in ``path``.
+
+    """
+    @classmethod
+    def init_extension(cls):
+        # wrapper which calls StandardsExportExtension.iso19115_export
+        def wrapper_entry(self: Entry, config_dict: dict):  
+            return StandardsExportExtension.iso19115_export(entry_or_resultset=self, config_dict=config_dict)
+        
+        # iso19115_export docstring and name for wrapper function
+        wrapper_entry.__doc__ = StandardsExportExtension.iso19115_export.__doc__
+        wrapper_entry.__name__ = StandardsExportExtension.iso19115_export.__name__
+        
+        # add wrapper to Entry model
+        Entry.export_iso19115 = wrapper_entry
+
+
+    @classmethod
+    def iso19115_export(cls, entry_or_resultset: Union[Entry, ImmutableResultSet], config_dict: dict):
+        """
+        Export a :class:`Entry <metacatalog.models.Entry>` or 
+        :class:`ImmutableResultSet <metacatalog.util.results.ImmutableResultSet>` to XML in 
+        ISO 19115 standard.
+        Repeatable information input is always a list, as we can loop over the lists in the
+        jinja ISO 19115 template.
+        Always returns an :class:`ElementTree <xml.etree.ElementTree.ElementTree>` object.
+
+        This function is added as a method to :class:`Entry <metacatalog.models.Entry>`
+        when the extension is activated. 
+
+        .. versionadded:: 0.7.7
+
+        Parameters
+        ----------
+        entry_or_resultset : Union[Entry, ImmutableResultSet]
+            The entry instance to be exported
+        config_dict : dict
+            Configuration dictionary, containing information about the data provider.
+            Mandatory (nested) keys and type of values:
+            * contact
+                * organisationName: str
+                * deliveryPoint: str
+                * city: str
+                * administrativeArea: str
+                * postalCode: str
+                * country: str
+                * electronicMailAddress: list(str)
+                * linkage: str
+                * linkage_name: str
+                * linkage_description: str
+            * publisher
+                * organisation_name: str
```

Comment:
you need blank lines between the different sections for sphinx to recognize the list. Better would be to add a code block:

```rst
.. code-block:: Python

  dict(
    organisationName: str = '',
    deliveryPoint: str = '',
  )
```

and have a blank line before and after the block definition